examples/js/loaders/GLTFLoader.js

```diff
@@ -704,6 +710,69 @@ THREE.GLTFLoader = ( function () {
 
 	};
 
+
+	/**
+	 * WebP Texture Extension
+	 *
+	 * Specification: https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Vendor/EXT_texture_webp
+	 */
+	function GLTFTextureWebPExtension( parser ) {
+
+		this.parser = parser;
+		this.name = EXTENSIONS.EXT_TEXTURE_WEBP;
+		this.isSupported = null;
+
+	}
+
+	GLTFTextureWebPExtension.prototype.loadTexture = function ( textureIndex ) {
+
+		var parser = this.parser;
+		var json = parser.json;
+
+		var textureDef = json.textures[ textureIndex ];
+
+		if ( ! textureDef.extensions || ! textureDef.extensions[ this.name ] || ! this.detectSupport() ) {
+
+			return null;
+
+		}
+
+		var extension = textureDef.extensions[ this.name ];
+		var source = json.images[ extension.source ];
+		var loader = source.uri ? parser.options.manager.getHandler( source.uri ) : parser.textureLoader;
+
+		return this.detectSupport()
+			.then( function () { return parser.loadTextureImage( textureIndex, source, loader ) } )
+			.catch( function () { return parser.loadTexture( textureIndex ); } );
```

Comment:
Here silently uses core spec texture as fallback for non-webp-support browser. Do you think if we should do console warnings for it like "Ignoring webp extension due to no webp support in this browser"?

Safari on old Mac seems the only browser in the major latest browsers which doesn't fully support WebP (except for IE) and I'm not sure how many users there are tho.

https://caniuse.com/webp

Reply:
Let's throw a clear error for `extensionRequired` case, yes.

When WebP is not supported but also not required, I think it's OK to use the fallback silently.